[docs] GCP MySQL IAM authentication support (#38943)
gravitational/teleport: docs/pages/database-access/guides/postgres-cloudsql.mdx

Teleport uses [IAM database authentication](https://cloud.google.com/sql/docs/postgres/authentication)
with Cloud SQL PostgreSQL instances.

(!docs/pages/includes/database-access/cloudsql_enable_iam_auth.mdx type="PostgreSQL" !)

> ## Step 2/7. Create a service account to access the PostgreSQL instance

Reviewer: ```suggestion
## Step 2/7. Create a service account for a database user
```

likewise with my comment from step 2/7 in mysql guide - if you decide to use my suggestion there
